internal/validitywindow/validitywindow_test.go

```diff
@@ -0,0 +1,292 @@
+			Containers: []container{
+				newContainer(1),
+			},
+			OldestAllowewd: 1,
+			ExpectedError:  nil,
+			ExpectedBits:   set.NewBits(0),
+		},
+		{
+			Name: "repeats after latest accepted block",
+			Blocks: []executionBlock{
+				newExecutionBlock(0, 0, 0, []int64{}),
+				newExecutionBlock(1, 1, 1, []int64{1}),
+				newExecutionBlock(2, 2, 2, []int64{2}),
+				newExecutionBlock(3, 3, 3, []int64{3}),
+				newExecutionBlock(4, 4, 4, []int64{4}),
+			},
+			Accepted:    1,
+			ParentBlock: newExecutionBlock(5, 5, 5, []int64{}),
+			Containers: []container{
+				newContainer(3),
+			},
+			OldestAllowewd: 1,
+			ExpectedError:  nil,
+			ExpectedBits:   set.NewBits(0),
+		},
+		{
+			Name: "missing block in ancestery",
+			Blocks: []executionBlock{
+				newExecutionBlock(0, 0, 0, []int64{}),
+				newExecutionBlock(1, 1, 1, []int64{1}),
+				newExecutionBlock(2, 2, 2, []int64{2}),
+			},
+			Accepted:    1,
+			ParentBlock: newExecutionBlock(5, 5, 5, []int64{}),
+			Containers: []container{
+				newContainer(3),
+			},
+			OldestAllowewd: 1,
+			ExpectedError:  database.ErrNotFound,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			r := require.New(t)
+			r.NoError(nil)
+
+			chainIndex := &testChainIndex{}
+			tvw := NewTimeValidityWindow(&logging.NoLog{}, trace.Noop, chainIndex)
+			r.NotNil(tvw)
+			for i, blk := range test.Blocks {
+				if i <= int(test.Accepted) {
+					tvw.Accept(blk)
+				}
+				chainIndex.set(blk.GetID(), blk)
+			}
+			bits, err := tvw.IsRepeat(context.Background(), test.ParentBlock, test.Containers, test.OldestAllowewd)
+			r.ErrorIs(err, test.ExpectedError)
+			if err == nil {
+				r.Equal(test.ExpectedBits.Bytes(), bits.Bytes())
+			}
+		})
+	}
+}
+
+type testChainIndex struct {
+	blocks map[ids.ID]ExecutionBlock[container]
+}
+
+func (t testChainIndex) GetExecutionBlock(_ context.Context, blkID ids.ID) (ExecutionBlock[container], error) {
+	if blk, ok := t.blocks[blkID]; ok {
+		return blk, nil
+	}
+	return nil, database.ErrNotFound
+}
+
+func (t *testChainIndex) set(blkID ids.ID, blk ExecutionBlock[container]) {
+	if t.blocks == nil {
+		t.blocks = make(map[ids.ID]ExecutionBlock[container])
+	}
+	t.blocks[blkID] = blk
+}
+
+type container struct {
+	ID     ids.ID
+	Expiry int64
+}
+
+func (c container) GetID() ids.ID {
+	return c.ID
+}
+
+func (c container) GetExpiry() int64 {
+	return c.Expiry
+}
+
+func newContainer(expiry int64) container {
+	return container{
+		Expiry: expiry,
+		ID:     int64ToID(expiry),
+	}
+}
+
+type executionBlock struct {
+	Prnt   ids.ID
+	Tmstmp int64
+	Hght   uint64
+	Ctrs   []container
+	ID     ids.ID
+}
+
+func (e executionBlock) GetID() ids.ID {
+	return e.ID
+}
+
+func (e executionBlock) GetParent() ids.ID {
+	return e.Prnt
+}
+
+func (e executionBlock) GetTimestamp() int64 {
+	return e.Tmstmp
+}
+
+func (e executionBlock) GetHeight() uint64 {
+	return e.Hght
+}
+
+func (e executionBlock) Containers() []container {
+	return e.Ctrs
+}
```

Comment:
If we are going to use `GetXXX`, for the rest of the interface required by validity window, could we do the same with `GetContainers()`?